rust/src/io/writer.rs

```diff
@@ -221,6 +228,26 @@ impl<'a> FileWriter<'a> {
         Ok(())
     }
 
+    async fn write_list_array(&mut self, field: &Field, array: &ArrayRef) -> Result<()> {
+        let list_arr = as_list_array(array);
+        let offsets: Int32Array = list_arr.value_offsets().iter().map(|o| *o).collect();
+        assert!(!offsets.is_empty());
+        let offsets = Arc::new(subtract_scalar(&offsets, offsets.value(0))?) as ArrayRef;
+        self.write_fixed_stride_array(field, &offsets).await?;
+        self.write_array(&field.children[0], &list_arr.values())
+            .await
+    }
+
+    async fn write_large_list_array(&mut self, field: &Field, array: &ArrayRef) -> Result<()> {
+        let list_arr = as_large_list_array(array);
+        let offsets: Int64Array = list_arr.value_offsets().iter().map(|o| *o).collect();
```

Comment:
does it not work if you just do `let offsets: &Int64Array = list_arr.value_offsets()` or something?

Reply:
`list_arr.value_offsets()` is a `&[i32]` array, not an Arrow Array.

Maybe i can just write the `&[i32]` array to disk directly?

Reply:
Oh, so offsets need to be computed to be relative offsets within the buffer again, which occurs a copy later.